acceptancetests/jujupy/wait_condition.py

```diff
@@ -321,6 +321,33 @@ def do_raise(self, model_name, status):
         raise AgentsNotStarted(model_name, status)
 
 
+class UnitInstallCondition(BaseCondition):
+
+    def __init__(self, unit, current, message, *args, **kwargs):
+        """Base condition for unit workload status."""
+        self.unit = unit
+        self.current = current
+        self.message = message
+        super(UnitInstallCondition, self).__init__(*args, **kwargs)
+
+    def iter_blocking_state(self, status):
+        """Wait until 'current' status and message matches supplied values."""
+        try:
+            unit = status.get_unit(self.unit)
+            unit_status = unit['workload-status']
+            cond_met = unit_status['current'] == self.current \
+                and unit_status['message'] == self.message
+        except KeyError:
+            cond_met = False
+        if not cond_met:
+            yield 'unit-workload', 'not-{}'.format(self.current)
+
+    def do_raise(self, model_name, status):
+        raise Exception(
```

Comment:
I imagine there will be (or are) other tests which poll status for a particular status/message. Should we introduce a bespoke exception class for that.